### PR TITLE
Add global for image registry

### PR DESCRIPTION
**Pull request by kfox1111**
This patch enables a user to override all image registry settings all at once to point to their own registry to enable easy custom mirrors.

partially implements: https://github.com/spiffe/helm-charts/issues/139

### CODE REVIEW

File: charts/spire/values.yaml [hunk 1]
```diff
@@ -11,6 +11,10 @@ global:
     # -- Override all instances of bundleConfigMap
     bundleConfigMap: ""
 
+    image:
+      # -- Override all images to point at your own registry
```
Comment (marcofranssen) [inline charts/spire/values.yaml:15]:
```suggestion
      # -- Override all Spire image registries at once
```

**Review by marcofranssen — COMMENTED** [review]
Doesn't necessarily have to be your own registry, would make the doc a bit more generic.

**Comment by faisal-memon** [comment]
This looks good to me. @marcofranssen what do you think?

**Review by marcofranssen — APPROVED** [review]
LGTM 🚀

**Pull request merged**